src/main/scala/org/broadinstitute/dsde/workbench/sam/api/SecurityDirectives.scala

```diff
@@ -51,6 +51,33 @@ trait SecurityDirectives {
       }
     }
 
+  def requireCreateWithParent(maybeParent: Option[FullyQualifiedResourceId], resourceType: ResourceType, userId: WorkbenchUserId, samRequestContext: SamRequestContext): Directive0 = {
```

Comment:
Just to understand this a bit, is this example correct?
I am a user, and I want to create a workspace with billing project Y as its parent. I might end up calling this method?

Reply:
well... users don't call methods, they call apis. But the api the user calls would go through this method.